### ISSUE TITLE
Multilingual typesetting with polyglossia results in Inverted Parentheses using algorithmic+algpsuedocode

### ISSUE THREAD

**Issue by ch0wner** [reutenauer/polyglossia#598]
I've already described my problem [here](https://tex.stackexchange.com/questions/687470/multilingual-typesetting-with-polyglossia-results-in-inverted-parentheses-using?noredirect=1#comment1705735_687470)

Do you have any suggestions?

**Comment by ch0wner** [comment]
I think I've just figured it out.. consider the following:
```
\begin{english}
	\begin{algorithm}[H]
		\caption{CREATE-P-SIZED-COUNTING-ARRAY PSUEDOCODE}
		\begin{algorithmic}[1]
			\Function{CREATE-P-SIZED-COUNTING-ARRAY}{A}
                \If{$length[A] = 0$} 					
                    \State \Return NIL \Comment{Edge-case when A is empty}
                \EndIf
                
                \State $n \gets length[A]$
                \State $p \gets A[1]$
                \For{$i \gets 2$ to $n$}
                    \State $A[i] \gets A[i] - n$
                    \If{$A[i] > p$}
                        \State $p \gets A[i]$
                    \EndIf
                \EndFor
                \LeftComment{Let c be a new array of size p, initialized with zeroes}
				\State \Return c
			\EndFunction
		\end{algorithmic}
	\end{algorithm}
\end{english}
```
![image](https://github.com/reutenauer/polyglossia/assets/32242613/23e8077f-f047-4fcf-a5bd-6ef7b92027eb)

But when I enclose the arguments in dollar signs
```
			\Function{CREATE-P-SIZED-COUNTING-ARRAY}{$A$}
```
The parentheses are flipped:
![image](https://github.com/reutenauer/polyglossia/assets/32242613/78994c94-5da3-49a5-b6d1-54692591480b)

Tested compiled using XeTex+MikTex and the David font for hebrew both on overleaf and LYX

**Comment by jspitz** [comment]
Please always provide a full minimal example case in the ticket, not only a snippet. 

Anyway, if I compile the MWE you gave at stackexchange, I (like others there) fail to reproduce:
![grafik](https://github.com/reutenauer/polyglossia/assets/4538783/a7438327-a64b-436c-b5b7-1efd876a5d15)

Like @Udi-Fogiel I suspect your `David CLM` font is defective.

**Comment by Udi-Fogiel** [comment]
I've tested on overleaf and there are no warnings about missing scripts. Maybe the problem is that the algorithm environment is a float, so the font switches from the english environment are not applied properly. With the following I get the correct result on overleaf

```TeX
\documentclass{article}

\usepackage{algorithm,algpseudocode}

\usepackage{polyglossia}
\usepackage{fontspec}
\setdefaultlanguage{hebrew}
\setotherlanguage{english}
\newfontfamily{\englishfont}{Latin Modern Roman}
\newfontfamily{\hebrewfont}[Script=Hebrew]{David CLM}
\newfontfamily{\hebrewfontsf}[Script=Hebrew]{Miriam CLM}
\newfontfamily{\hebrewfonttt}[Script=Hebrew]{Miriam Mono CLM}

\begin{document}
\section{אלגוריתם לדוגמה}


\begin{algorithm}
\begin{english} % <---- english inisde algorithm
    \caption{Euclid’s algorithm}
    \label{euclid}
    \begin{algorithmic} 
        \Procedure{Euclid}{$a,b$} \Comment{The g.c.d. of a and b}
            \State $r\gets a \bmod b$
            \While{$r\not=0$} \Comment{We have the answer if r is 0}
                \State $a \gets b$
                \State $b \gets r$
                \State $r \gets a \bmod b$
            \EndWhile\label{euclidendwhile}
            \State \textbf{return} $b$\Comment{The gcd is b}
        \EndProcedure
    \end{algorithmic}
    \end{english}
\end{algorithm}

\end{document}
```  

I don't know why on an updated version it works ok, maybe something in floats has changed in 2023.

**Comment by jspitz** [comment]
As to float, d4ae868f4ad154499b4 (as of v. 1.58 of `polyglossia`) might be relevant.

Which version of `polyglossia` does overleaf have?

**Comment by Udi-Fogiel** [comment]
> Which version of `polyglossia` does overleaf have?

polyglossia 2022/04/20 v1.56

Tried to "apply" the patch in overleaf, did not change anything, 
and no errors.

**Comment by jspitz** [comment]
> As to float, [d4ae868](https://github.com/reutenauer/polyglossia/commit/d4ae868f4ad154499b48f4ab69e317424e210628) (as of v. 1.58 of `polyglossia`) might be relevant (see #542)..

Bisect points at 84bd1b4eaf6. So we can close this as fixed as of version 1.61.
